bentoml/deployment/serverless/aws_lambda.py

```diff
@@ -0,0 +1,229 @@
+
+import os
+import logging
+from packaging import version
+
+from ruamel.yaml import YAML
+
+from bentoml.utils import Path
+from bentoml.deployment.operator import DeploymentOperatorBase
+from bentoml.yatai.status import Status
+from bentoml.exceptions import BentoMLDeploymentException, BentoMLException
+from bentoml.proto.deployment_pb2 import (
+    Deployment,
+    ApplyDeploymentResponse,
+    DescribeDeploymentResponse,
+    DeleteDeploymentResponse,
+    DeploymentState,
+)
+from bentoml.deployment.serverless.serverless_utils import (
+    call_serverless_command,
+    install_serverless_plugin,
+    TemporaryServerlessContent,
+    TemporaryServerlessConfig,
+    parse_serverless_info_response_to_json_string,
+)
+from bentoml.archive.loader import load_bentoml_config
+
+logger = logging.getLogger(__name__)
+
+AWS_HANDLER_PY_TEMPLATE_HEADER = """\
+try:
+    import unzip_requirements
+except ImportError:
+    pass
+
+from {class_name} import {class_name}
+
+bento_service = {class_name}.load()
+
+"""
+
+AWS_FUNCTION_TEMPLATE = """\
+def {api_name}(event, context):
+    api = bento_service.get_service_api('{api_name}')
+
+    return api.handle_aws_lambda_event(event)
+
+"""
+
+
+def generate_aws_handler_functions_config(apis):
+    function_list = {}
+    for api in apis:
+        function_list[api['name']] = {
+            "handler": "handler." + api['name'],
+            "events": [{"http": {"path": "/" + api['name'], "method": "post"}}],
+        }
+    return function_list
+
+
+def generate_serverless_configuration_for_aws_lambda(
+    service_name, apis, output_path, region, stage
+):
+    config_path = os.path.join(output_path, "serverless.yml")
+    yaml = YAML()
+    with open(config_path, "r") as f:
+        content = f.read()
+    serverless_config = yaml.load(content)
+
+    serverless_config["service"] = service_name
+    serverless_config["provider"]["region"] = region
+    logger.info("Using user AWS region: %s", region)
+
+    serverless_config["provider"]["stage"] = stage
+    logger.info("Using AWS stage: %s", stage)
+
+    serverless_config["functions"] = generate_aws_handler_functions_config(apis)
+
+    serverless_config["custom"] = {
+        "apigwBinary": ["image/jpg", "image/jpeg", "image/png"],
+        "pythonRequirements": {
+            "useDownloadCache": True,
+            "useStaticCache": True,
+            "dockerizePip": True,
+            "slim": True,
+            "strip": True,
+            "zip": True,
+        },
+    }
+
+    yaml.dump(serverless_config, Path(config_path))
+    return
+
+
+def generate_handler_py(bento_name, apis, output_path):
+    file_content = [AWS_HANDLER_PY_TEMPLATE_HEADER.format(class_name=bento_name)]
+
+    for api in apis:
+        api_content = AWS_FUNCTION_TEMPLATE.format(api_name=api['name'])
+        file_content.append(api_content)
+
+    with open(os.path.join(output_path, "handler.py"), "w") as f:
+        f.write(''.join(file_content))
```

Comment:
Sorry for fluctuating...but it looks better to just to write to file for each API for a better balance of IO and CPU resources...